### PR TITLE
Fix pmi warnings

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Remove some unused labels.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/pmi

File: src/pmi/src/pmi_wire.c [hunk 1]
```diff
@@ -606,18 +604,10 @@ int PMIU_cmd_output_v1(struct PMIU_cmd *pmicmd, char **buf_out, int *buflen_out)
 
     *buf_out = pmicmd->tmp_buf;
     *buflen_out = buflen;
-
-  fn_exit:
-    return pmi_errno;
-
-  fn_fail:
-    goto fn_exit;
```
Comment (hzhou) [inline src/pmi/src/pmi_wire.c:614]: Why don't we just remove the labels? The function may fail, for example, out of memory or exceeding temp buffer limit or `pmicmd` fails validation. Even though we do not check these failures now, having the error return makes future modification easier.

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**